DigitalLearningSolutions.Web/Controllers/FrameworksController/Frameworks.cs

```diff
@@ -1,555 +1,659 @@
-            var model = new FrameworkViewModel()
-            {
-                DetailFramework = detailFramework
-            };
-            if (tabname == "Details")
-            {
-                model.Collaborators = frameworkService.GetCollaboratorsForFrameworkId(frameworkId);
-                model.FrameworkDefaultQuestions = frameworkService.GetFrameworkDefaultQuestionsById(frameworkId, adminId);
-            }
-            if (tabname == "Structure")
-            {
-                model.FrameworkCompetencyGroups = frameworkService.GetFrameworkCompetencyGroups(frameworkId).ToList();
-                model.FrameworkCompetencies = frameworkService.GetFrameworkCompetenciesUngrouped(frameworkId);
-            }
-            if (tabname == "Comments")
-            {
-                model.CommentReplies = frameworkService.GetCommentsForFrameworkId(frameworkId, adminId);
-            }
-            return View("Developer/Framework", model);
-        }
-        public IActionResult InsertFramework()
-        {
-            var adminId = GetAdminID();
-            SessionNewFramework sessionNewFramework = TempData.Peek<SessionNewFramework>();
-            DetailFramework? detailFramework = sessionNewFramework.DetailFramework;
-            detailFramework = InsertBrandingCategoryTopicIfRequired(detailFramework);
-            if (detailFramework == null || adminId < 1)
-            {
-                logger.LogWarning($"Failed to create framework: adminId: {adminId}");
-                return StatusCode(500);
-            }
-            var newFramework = frameworkService.CreateFramework(detailFramework, adminId);
-            TempData.Clear();
-            return RedirectToAction("AddCollaborators", "Frameworks", new { actionname = "New", frameworkId = newFramework.ID });
-        }
-
-    }
-}
+﻿namespace DigitalLearningSolutions.Web.Controllers.FrameworksController
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models.Frameworks;
+    using DigitalLearningSolutions.Data.Models.SessionData.Frameworks;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common;
+    using DigitalLearningSolutions.Web.ViewModels.Frameworks;
+    using DigitalLearningSolutions.Web.ViewModels.Frameworks.Dashboard;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Rendering;
+    using Microsoft.Extensions.Logging;
+
+    public partial class FrameworksController
+    {
+        private const string CookieName = "DLSFrameworkService";
+
+        [Route("/Frameworks/MyFrameworks/{page=1:int}")]
+        public IActionResult FrameworksDashboard(
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.DescendingText,

```

Comment:
All changes in this file are either changing the sort direction string, the sort by property, or the SortingHelper